### PR TITLE
backend: transition to @safe code part 7

### PR DESCRIPTION
**Pull request by WalterBright**
Rationale is here: #12327

### CODE REVIEW

File: src/dmd/backend/mscoffobj.d [hunk 1]
```diff
@@ -409,19 +414,21 @@ version (SCPP)
...
+@trusted
+int32_t *MsCoffObj_patchAddr(int seg, targ_size_t offset)
 {
     return cast(int32_t *)(fobjbuf.buf + ScnhdrTab[SegData[seg].SDshtidx].PointerToRawData + offset);
 }
 
-private extern (D)
-int32_t *patchAddr64(int seg, targ_size_t offset)
+@trusted
+int32_t *MsCoffObj_patchAddr64(int seg, targ_size_t offset)
```
Comment (RazvanN7) [inline src/dmd/backend/mscoffobj.d:424]: Why is `private`  removed here?

Comment (RazvanN7) [reply src/dmd/backend/mscoffobj.d:424]: Also, why is this also changing the name of the function. Someone once said that refactorings should not be mixed with additions :P .

Comment (WalterBright) [reply src/dmd/backend/mscoffobj.d:425]: The name was changed so it did not collide with one in machobj.d

**Comment by WalterBright** [comment]
The C++ interop error looks spurious and unrelated.

File: src/dmd/backend/mscoffobj.d [hunk 2]
```diff
@@ -409,19 +414,22 @@ version (SCPP)
  * pseg/offset to start of seg.
  */
 
-private extern (D)
-int32_t *patchAddr(int seg, targ_size_t offset)
+@trusted
+int32_t *MsCoffObj_patchAddr(int seg, targ_size_t offset)
```
Comment (RazvanN7) [inline src/dmd/backend/mscoffobj.d:418]: `private` has been removed here also

File: src/dmd/backend/mscoffobj.d [hunk 3]
```diff
@@ -409,19 +414,22 @@ version (SCPP)
...
-int32_t *patchAddr64(int seg, targ_size_t offset)
+int32_t *MsCoffObj_patchAddr64(int seg, targ_size_t offset)
 {
     return cast(int32_t *)(fobjbuf.buf + ScnhdrTab[SegData[seg].SDshtidx].PointerToRawData + offset);
 }
 
-private extern (D)
+static if (0)
+{
+@trusted
```
Comment (RazvanN7) [inline src/dmd/backend/mscoffobj.d:432]: `private` has been removed here also.

File: src/dmd/backend/mscoffobj.d (hunk 2 above)
Comment (WalterBright) [reply src/dmd/backend/mscoffobj.d:419]: done

File: src/dmd/backend/mscoffobj.d (hunk 3 above)
Comment (WalterBright) [reply src/dmd/backend/mscoffobj.d:434]: done

**Review by Geod24 — APPROVED** [review]

**Pull request merged**